pkg/delete/delete.go

```diff
@@ -71,7 +70,7 @@ func Configs(ctx context.Context, clients ClientSet, apis api.APIs, automationRe
 			deleteErrs := deleteClassicConfig(ctx, clients.Classic, targetApi, entries, entryType)
 			errs = append(errs, deleteErrs...)
 		} else if targetAutomation, isAutomation := automationResources[entryType]; isAutomation {
-			if clients.Automation == nil {
+			if reflect.ValueOf(clients.Automation).IsNil() {
```

Comment:
I personally dislike this, but there is no clean way to check the underlying object behind the interface for nil. The interface is never nil, it just happens to contain a nil pointer object.

The cleanest option would be to make the Automation client's zero/nil value useful and make it just do nothing. That would need us to change the client's method to pointer receivers, make them check for nil and then possible do nothing. 

That might be a good pattern to follow for the -core clients. But I think it's too much for this change, esp. as we want to move the automation client to -core soon. 
(Why I decided against the option to just produce an error earlier if there are 3rd gen types but no oauth, see the commit message. TL;DR: it seems more userfriendly.)